### PR TITLE
Add conversation consensus functionality

### PR DESCRIPTION
**Pull request by whsieh**
Allow users to edit and approve resolutions and summaries.

### CODE REVIEW

File: app/assets/javascripts/homepage/conversation.coffee [hunk 1]
```diff
@@ -95,10 +161,28 @@ conversation.controller("ConversationController", ["$scope", "$stateParams", "AP
...
+                delete post["post_type"]
+                postsById[post.id] = post
+            if response.summaries[0].author.id == AppState.user.id
+                selfIndex = 0
+            else if response.summaries[1].author.id == AppState.user.id
+                selfIndex = 1
+            conversation.pendingSummaries.own = response.summaries[1 - selfIndex].text
+            conversation.pendingSummaries.opposing = response.summaries[selfIndex].text
+            conversation.opponent = response.summaries[1 - selfIndex].author
+            conversation.self = response.summaries[selfIndex].author
```
Comment (leohtkam) [inline app/assets/javascripts/homepage/conversation.coffee]: Should leave a comment here explaining which is which.

File: app/assets/javascripts/homepage/conversation.coffee [hunk 2]
```diff
@@ -95,10 +161,28 @@ conversation.controller("ConversationController", ["$scope", "$stateParams", "AP
...
+            for post in $scope.posts
+                post.type = post.post_type
+                delete post["post_type"]
+                postsById[post.id] = post
+            if response.summaries[0].author.id == AppState.user.id
+                selfIndex = 0
+            else if response.summaries[1].author.id == AppState.user.id
+                selfIndex = 1
+            conversation.pendingSummaries.own = response.summaries[1 - selfIndex].text
+            conversation.pendingSummaries.opposing = response.summaries[selfIndex].text
```
Comment (leohtkam) [inline app/assets/javascripts/homepage/conversation.coffee:185]: Should leave a comment here explaining what are "own" and "opposing"

File: app/models/conversation.rb [hunk 3]
```diff
@@ -17,12 +17,93 @@ def timestamp
...
             :title => title,
+            :summaries => [
+                {
+                    :author => arena.user2.response_object,
+                    :text => self.summary_of_first
+                },
+                {
+                    :author => arena.user1.response_object,
+                    :text => self.summary_of_second
+                }
```
Comment (leohtkam) [inline app/models/conversation.rb:35]: Might also want a comment here.

File: app/models/conversation.rb [hunk 4]
```diff
@@ -17,12 +17,93 @@ def timestamp
...
+        self.resolution_updated_at = Time.now
+        self.resolution_updated_by_id = user_id
+        save
+        return true
+    end
+
+    def user_did_approve_resolution(user_id)
+        if allow_user_to_approve_consensus user_id
+            self.resolution_state = :consensus_reached
+            posts.create :author => arena.user1, :text => resolution, :post_type => :resolution
```
Comment (leohtkam) [inline app/models/conversation.rb:94]: Resolution post always show the first user as the author?

**Comment by leohtkam** [comment]
+1 Looks great!
Though missing cucumber tests.

File: app/models/conversation.rb (hunk 4 above)
Comment (whsieh) [inline app/models/conversation.rb:94]: The post is "authored" under user1, but that's not what's displayed. What's displayed to the user is:
You and ____:
x hours ago
Lorem ipsum.........

File: app/assets/javascripts/homepage/conversation.coffee (hunk 2 above)
Comment (whsieh) [inline app/assets/javascripts/homepage/conversation.coffee:185]: Addressed with the latest commit.

File: app/models/conversation.rb (hunk 3 above)
Comment (whsieh) [inline app/models/conversation.rb:35]: Addressed with the latest commit.

**Comment by nickfong** [comment]
+1

I'll merge in to `iter4` shortly (even without Cuke) since we're pushing for getting everything hooked up ASAP

**Pull request merged**